### DISCUSSION TITLE
Custom spinners

### DISCUSSION THREAD

**Discussion by willhansen** [helix-editor/helix#7828] [General]
I'm tired of the default braille LSP spinner.  I want a custom one.  

Just a configuration option that lets me input a list of strings to cycle through.

- I'd settle for one character spinners

- Better would be arbitrary length spinners (allowing stuff like a braille dot doing a figure-8 through 3 braille characters)

- Even better would be somehow incorporating colors into it.  (Maybe similar to how colors are printed in zsh themes?)

Looks not too difficult to implement (if I'm not already missing an existing configuration option for it)

Relevant code I could find:
- https://github.com/helix-editor/helix/blob/bc737404e8ad462b4101987730f4a76658d132ad/helix-term/src/ui/spinner.rs#L14 
- https://github.com/helix-editor/helix/blob/bc737404e8ad462b4101987730f4a76658d132ad/helix-term/src/ui/spinner.rs#L49 
- https://github.com/helix-editor/helix/blob/bc737404e8ad462b4101987730f4a76658d132ad/helix-term/src/ui/editor.rs#L73 

Might implement it myself.

**Comment by woojiq** [comment]
_OT_: I'd suggest you set `lsp.display-messages` to `true` and you won't need the spinner anymore.

**Comment by willhansen** [comment]
I already have that, but I find the spinner useful as a "thinking..." indicator.  Also for aesthetic reasons.

**Comment by pascalkuthe** [comment]
I don't think these need to be configured . The spinner is just supposed to indicate that work is being done in the background (so it's clear the server is still working and hasn't crashed). The spinners are not even continuously animated) so it probably wound't look that good anyway. 

We try to avoid adding too many config options and in this case the default is good enough to hardcode IMO.

**Comment by willhansen** [comment]
I can certainly agree it's not worth it if the spinner isn't even continuous.

Was that a design decision, or just not worth the effort?

**Comment by pascalkuthe** [comment]
https://github.com/helix-editor/helix/pull/6446#issuecomment-1488071400

Not worth the extra overhead and battery life cost. This could be made more performant by allowing partial invalidation in the rendering cise but that's a lot of work
